src/rgw/rgw_sal.h

```diff
@@ -328,6 +328,23 @@ class Driver {
                                 RGWObjVersionTracker* objv_tracker,
                                 optional_yield y,
                                 const DoutPrefixProvider* dpp) = 0;
+    /* Update the bucket-topic mapping in the omap, if |add_mapping|=true then
+     * adding the |bucket_str| to omap, else delete the |bucket_str| from the
+     * omap.  */
+    virtual int update_bucket_topic_mapping(const rgw_pubsub_topic& topic,
+                                            rgw::sal::Bucket* bucket,
+                                            bool add_mapping,
+                                            optional_yield y,
+                                            const DoutPrefixProvider* dpp) = 0;
+    /** Get the bucket-topic mapping from the omap */
+    virtual int get_bucket_topic_mapping(const rgw_pubsub_topic& topic,
+                                         std::set<std::string>& bucket_keys,
+                                         optional_yield y,
+                                         const DoutPrefixProvider* dpp) = 0;
+    /** Remove the bucket-topic mapping omap */
+    virtual int delete_bucket_topic_omap(const rgw_pubsub_topic& topic,
```

Comment:
i think that "omap" is rados specific, better to use the term "mapping" here as welll

Reply:
> i think that "omap" is rados specific, better to use the term "mapping" here as welll

Done